ballista/rust/scheduler/src/lib.rs

```diff
@@ -116,30 +117,29 @@ pub struct SchedulerServer<T: 'static + AsLogicalPlan, U: 'static + AsExecutionP
     pub(crate) state: Arc<SchedulerState<T, U>>,
     start_time: u128,
     policy: TaskSchedulingPolicy,
-    scheduler_env: Option<SchedulerEnv>,
+    scheduler_loop: Option<SchedulerLoop>,
     executors_client: Option<ExecutorsClient>,
-    ctx: Arc<RwLock<ExecutionContext>>,
     codec: BallistaCodec<T, U>,
+    /// DataFusion session contexts that are registered within the SchedulerServer
+    session_context_registry: Arc<SessionContextRegistry>,
```

Comment:
So I think we need to preserve some mechanism to register common extensions in the `Context`. The original intention of adding `ExecutionContext` to the `SchedulerServer` was to allow creating a `SchedulerServer` which had a customized `ExecutionContext`. Looks like you've dealt with custom `ObjectStore` implementations by moving that to `RuntimeEnv` but there are serveral other extension points which may be useful such as custom planners, optimizers, udf/udaf 

Maybe we can add a "default" `SessionContext` to `SessionContextRegistry` so we can create new contexts from a template? That way the template context can be initialized with custom extensions.

Reply:
To add on to that slightly, previous work allows the serialization of extension plans in Ballista, but in order to actually use them an `ExtensionPlanner` needs to be registered in the `DefaultQueryPlanner`.